crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -805,6 +805,53 @@ impl<P: RoomDataProvider> TimelineInner<P> {
         }
     }
 
+    pub(super) async fn replace_local_echo(
+        &self,
+        txn_id: &TransactionId,
+        content: AnyMessageLikeEventContent,
+    ) -> bool {
+        let AnyMessageLikeEventContent::RoomMessage(content) = content else {
+            // Ideally, we'd support replacing local echoes for a reaction, etc., but
+            // handling RoomMessage should be sufficient in most cases. Worst
+            // case, the local echo will be sent Soon™ and we'll get another chance at
```

Comment:
“Soon™” 😄  

By the way, I agree with this. It means someone reacts to its own local event, and wants to change that. That's clearly an edge case.